### PR TITLE
null values filtered from api response

### PR DESCRIPTION
**Pull request by cs-pranali-chavan**
null values filtered from api response
null filter for added in api payload

### CODE REVIEW

File: widget/widgetAssets/js/socOverviewSankey.service.js [hunk 1]
```diff
@@ -110,42 +110,57 @@
...
-              } else {
-                  dataFilters = frontFilter;
-              }
-  
-              return dataFilters;
-          }
+            queryObject.aggregates.push({
+                'operator': 'groupby',
+                'alias': 'series_' + elementIndex,
+                'field': currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany' ? currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.itemValue' : currentLayer['targetNodeSubField']
```
Comment (cs-vikas-shelar) [inline widget/widgetAssets/js/socOverviewSankey.service.js]: Here in else condition of ternary operator it should be currentLayer['targetNodeField'].currentLayer['targetNodeSubField']

File: widget/widgetAssets/js/socOverviewSankey.service.js [hunk 2]
```diff
@@ -110,42 +110,57 @@
...
+
+
+        function getFilters(duration) {
+            let frontFilter = {};
+            frontFilter.filters = [];
+            for (var i = 0; i < config.layers.length; i++) {
+                let currentLayer = config.layers[i];
+                let _nullField = '';
+                if(currentLayer['targetNodeSubField'] !== ''){
+                    _nullField = currentLayer['targetNodeSubType'] === 'picklist' || currentLayer['targetNodeSubType'] === 'manyToMany' ? currentLayer['targetNodeField'] + '.' + currentLayer['targetNodeSubField'] + '.itemValue' : currentLayer['targetNodeSubField'];
```
Comment (cs-vikas-shelar) [inline widget/widgetAssets/js/socOverviewSankey.service.js]: Same here in the else condition of the ternary operator it should be currentLayer['targetNodeField'].currentLayer['targetNodeSubField']

**Review by cs-vikas-shelar — CHANGES_REQUESTED** [review]

File: widget/edit.controller.js [hunk 3]
```diff
@@ -203,9 +203,9 @@
               LABEL_FILTER_CRITERIA: widgetUtilityService.translate('socOverviewSankey.LABEL_FILTER_CRITERIA'),
               LABEL: widgetUtilityService.translate('socOverviewSankey.LABEL'),
               LABEL_SOURCE_NODE: widgetUtilityService.translate('socOverviewSankey.LABEL_SOURCE_NODE'),
-              LABEL_SELECT_SOURCE_NODES: widgetUtilityService.translate('socOverviewSankey.LABEL_SELECT_SOURCE_NODES'),
+              LABEL_SELECT_SOURCE_NODE: widgetUtilityService.translate('socOverviewSankey.LABEL_SELECT_SOURCE_NODE'),
```
Comment (cs-vikas-shelar) [inline widget/edit.controller.js:208]: I don't see any HTML file changes where these variables are used. Changes might required in those files too.

Comment (cs-pranali-chavan) [reply widget/edit.controller.js:208]: only changes were not updated in edit controller

**Comment by cs-vikas-shelar** [comment]
Code changes look fine. Please check for the date-time scenario once.

**Comment by cs-vikas-shelar** [comment]
Code changes look fine.

A minor comment that can be addressed later:
'selectedTargetNodeFields' and 'extendedTargetNodeFields' are not required to be part of $scope.params as those are not needed for calculations later. It can be used as a local variable or a statement can be optimized.

**Pull request merged**